src/main/java/SYSC4806Project/ItemQuantityList.java

```diff
@@ -40,7 +43,7 @@ public boolean addItems(Product product, int quantity) {
      * @return true if the product was added
      */
     public boolean addProduct(Product product) {
-        if (product == null || !this.contains(product)) {return false;}
```

Comment:
@Rebecca-Elliott From testing, I think that was just a flipped condition - removed the `!`. I think the check should be to see if the `Product` is `null` or is already in the `ItemQuantityList`. Can you check this makes sense? Thanks!

Reply:
yes it was a typo.